### PR TITLE
Moves String.Comparison.cs to shared location

### PR DESCRIPTION
**Pull request by marek-safar**

### CODE REVIEW

File: src/mscorlib/src/System/String.CoreCLR.cs [hunk 1]
```diff
@@ -64,6 +64,18 @@ public partial class String
...
+        [MethodImplAttribute(MethodImplOptions.InternalCall)]
+        internal static extern int CompareOrdinalHelper(String strA, int indexA, int countA, String strB, int indexB, int countB);
+
+        //This will not work in case-insensitive mode for any character greater than 0x7F.
+        //We'll throw an ArgumentException.
+        [MethodImplAttribute(MethodImplOptions.InternalCall)]
+        internal static extern unsafe int nativeCompareOrdinalIgnoreCaseWC(String strA, sbyte* strBBytes);
+
+        [MethodImplAttribute(MethodImplOptions.InternalCall)]
+        private static extern int InternalMarvin32HashString(string s);
```
Comment (jkotas) [inline src/mscorlib/src/System/String.CoreCLR.cs]: I am fixing this one in https://github.com/dotnet/coreclr/pull/17029.

Comment (marek-safar) [reply src/mscorlib/src/System/String.CoreCLR.cs]: I noticed. I think moving whole GetHashCode (+deps) to String.cs is best here

Comment (jkotas) [reply src/mscorlib/src/System/String.CoreCLR.cs]: Why? It is same between CoreCLR and CoreRT.

File: src/mscorlib/shared/System/String.Comparison.cs [hunk 2]
```diff
@@ -61,21 +57,6 @@ private static unsafe int CompareOrdinalIgnoreCaseHelper(String strA, String str
...
         }
 
-        // native call to COMString::CompareOrdinalEx
-        [MethodImplAttribute(MethodImplOptions.InternalCall)]
-        internal static extern int CompareOrdinalHelper(String strA, int indexA, int countA, String strB, int indexB, int countB);
-
-        //This will not work in case-insensitive mode for any character greater than 0x7F.  
-        //We'll throw an ArgumentException.
-        [MethodImplAttribute(MethodImplOptions.InternalCall)]
-        internal static extern unsafe int nativeCompareOrdinalIgnoreCaseWC(String strA, sbyte* strBBytes);
```
Comment (jkotas) [inline src/mscorlib/shared/System/String.Comparison.cs]: Taking care of this one in #17061

**Comment by jkotas** [comment]
There will be more changes necessary to make the shared implementation build and work well under both CoreCLR and CoreRT.

**Comment by marek-safar** [comment]
The shared implementation has only managed code and it should work everywhere. CoreRT already has `CompareOrdinalHelper` but probably missing some of the dependencies. Mono would like to use the shared implementation and CoreRT can switch to it once it's ready

**Comment by danmosemsft** [comment]
@jkotas what remains here?

**Comment by jkotas** [comment]
It needs to build and work in CoreRT/ProjectN. I have been working through the issues to make it work well. I think we can close this one since it has conflicts. I will take care of submitting PR with the required changes.

**Comment by jkotas** [comment]
Superseded by https://github.com/dotnet/coreclr/pull/17247

**Pull request closed without merging**